fastFM/tests/test_ffm.py

```diff
@@ -38,5 +52,22 @@ def test_ffm2_predict_w0():
     y_pred = ffm2.ffm_predict(w0, w, V, X)
     assert_equal(y_pred, w0)
 
-if __name__ == '__main__':
-    test_ffm2_predict()
+def test_ffm2_fit():
+    w0, w, V, y, X = get_another_test_problem()
+    w0_init = w0
+    w_init = np.copy(w)
+    V_init = np.copy(V)
+    rank = 2
+
+    y_pred = ffm2.ffm_predict(w0, w, V, X)
+    msqr_before = mean_squared_error(y, y_pred)
+
+    w0, w, V = ffm2.ffm_als_fit(w0, w, V, X, y, rank)
```

Comment:
Should be called with init values `ffm_als_fit(w0_init, ...)`.

Reply:
They will be changed in-place...

Reply:
`w0_init, w_int, V_init = ffm_als_fit(w0_init, ...)`? I was aiming for a clear distinction between the parameters belonging to the test example and the ones we optimize.

Reply:
BTW if everything get's changed in-place then we shouldn't return anything?

Reply:
arrays by pointers and memory views, so except w0